core/src/tokenizer.rs

```diff
@@ -1027,7 +1076,9 @@ impl<'a> Tokenizer<'a> {
             self.bump();
         }
 
-        return LiteralInteger(value);
+        Value::Number(unsafe {
+            OwnedSlice::from_str(&value.to_string())
```

Comment:
`OwnedSlice::from_str` is explicitly an unsafe function. It works only if the `&str` passed to it outlives the slice. In this case you are creating an owned `String` which lives only until the end of `read_hexadec`, while the `OwnedSlice` created out of it continues and the raw pointer inside of it is now pointing to garbage.

Need a better solution (as mentioned on Gitter).